### PR TITLE
[@types/roslib] Add missing options to main constructor.

### PR DESCRIPTION
**Pull request by VorpalBlade**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: http://robotwebtools.org/jsdoc/roslibjs/current/Ros.html
- [X] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [X] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/roslib/index.d.ts [hunk 1]
```diff
@@ -29,10 +31,16 @@ declare namespace ROSLIB {
...
+		 *   * url (optional) - (can be specified later with `connect`) the WebSocket URL for rosbridge or the node server url to connect using socket.io (if socket.io exists in the page) <br>
+		 *   * groovyCompatibility - don't use interfaces that changed after the last groovy release or rosbridge_suite and related tools (defaults to true)
+		 *   * transportLibrary (optional) - one of 'websocket', 'workersocket' (default), 'socket.io' or RTCPeerConnection instance controlling how the connection is created in `connect`.
+		 *   * transportOptions (optional) - the options to use use when creating a connection. Currently only used if `transportLibrary` is RTCPeerConnection.
 		 */
 		constructor(options:{
-			url?: string
+			url?: string,
+			groovyCompatibility?: boolean,
+			transportLibrary?: 'websocket' | 'wokersocket' | 'socket.io' | RTCPeerConnection,
```
Comment (skycoop) [inline types/roslib/index.d.ts]: `workersocket` is not listed as an option in the [original docstring](https://github.com/RobotWebTools/roslibjs/blob/1.0.1/src/core/Ros.js#L29)

```suggestion
			transportLibrary?: 'websocket' | 'socket.io' | RTCPeerConnection,
```

File: types/roslib/index.d.ts [hunk 2]
```diff
@@ -29,10 +31,16 @@ declare namespace ROSLIB {
 		 *
 		 * @constructor
 		 * @param options - possible keys include:
-		 *   * url (optional) - the WebSocket URL for rosbridge (can be specified later with `connect`)
+		 *   * url (optional) - (can be specified later with `connect`) the WebSocket URL for rosbridge or the node server url to connect using socket.io (if socket.io exists in the page) <br>
+		 *   * groovyCompatibility - don't use interfaces that changed after the last groovy release or rosbridge_suite and related tools (defaults to true)
+		 *   * transportLibrary (optional) - one of 'websocket', 'workersocket' (default), 'socket.io' or RTCPeerConnection instance controlling how the connection is created in `connect`.
```
Comment (skycoop) [inline types/roslib/index.d.ts]:
```suggestion
		 *   * transportLibrary (optional) - one of 'websocket' (default), 'socket.io' or RTCPeerConnection instance controlling how the connection is created in `connect`.
```

**Review by skycoop — CHANGES_REQUESTED** [review]
Looks good to me other than a slight mismatch on the possible values for `transportLibrary`

**Comment by VorpalBlade** [comment]
> Looks good to me other than a slight mismatch on the possible values for `transportLibrary`

Ah, that might be because I'm using the latest git revision. It has been added since. Given that it will presumably be valid in the next release, should we keep it or remove it?

**Comment by skycoop** [comment]
My personal inclination is to match the current released version of the package (v1.0.1) since:
- That's what the version comment has listed
- To avoid unnecessary confusion we should probably try to adhere to the currently live [JSDocs](http://robotwebtools.org/jsdoc/roslibjs/current/Ros.html)

**Review by skycoop — APPROVED** [review]
Thanks for making those changes :grin:

**Pull request merged**